### ISSUE TITLE
`File.Select.files` fails on iOS 12 browsers.

### ISSUE THREAD

**Issue by kyasu1** [elm/file#4]
The examples using `File.Select.files` fail to call its callback on iOS 12 browsers. With my iPhone Xs/5s, both 'Take Photo' and choosing from 'Photo Library' did not work at all. While with my iPad Pro, 'Take Photo' did not work but choosing from 'Photo Library' worked. 

I think this issue is related to the problem reported on [this Discourse topic](https://discourse.elm-lang.org/t/elm-file-selecting-files-using-file-select-file-s-sometimes-fails/2701). Applying the fix suggested at the last post (moving the `node` variable declaration to the outer scope) to the generated JavaScript file seems to solve the problem. 

This API is really useful when combining with `elm-ui`, hope this will be resolved soon !

**Comment by charukiewicz** [comment]
Confirming the same issue.

Testing using Safari on an old iPhone 6 (iOS v10), both 'Take Photo' and 'Photo Library' fail every time.

On Chrome 71 on my Android (version 8.0.0), 'Use Camera' seems to be mostly reliable (failing occasionally), and 'Select File' seems to be 100% reliable.

**Comment by charukiewicz** [comment]
I just did some testing based off of what @evancz suggested in the Discourse topic, to quote him:

>Can someone test this with just JS code to see:
>
>1. If the bug reproduces with the exact same JS and no Elm code.
>2. If so, does adding it to the DOM and then removing it in the event handler make it more reliable?

The existing version of the [`Elm.Kernel.File`](https://github.com/elm/file/blob/1f913d8682ad9ab6eed930691d41c71a9c0b8d72/src/Elm/Kernel/File.js#L82-L95) function is:

```javascript
function _File_uploadOne(mimes)
{
	return __Scheduler_binding(function(callback)
	{
		var node = document.createElementNS('http://www.w3.org/1999/xhtml', 'input');
		node.setAttribute('type', 'file');
		node.setAttribute('accept', A2(__String_join, ',', mimes));
		node.addEventListener('change', function(event)
		{
			callback(__Scheduler_succeed(event.target.files[0]));
		});
		node.dispatchEvent(new MouseEvent('click'));
	});
}
```

However, I made a quick-and-dirty modification to make the function append the node to the DOM and then remove it:

```javascript
function _File_uploadOne(mimes)
{
    return _Scheduler_binding(function(callback)
    {
        var node = document.createElement('input');
        node.setAttribute('type', 'file');
        node.setAttribute('accept', A2(elm$core$String$join, ',', mimes));
        node.addEventListener('change', function(event)
        {  
            document.body.removeChild(node); // node is removed after change event fires
            callback(_Scheduler_succeed(event.target.files[0]));
        });   
        document.body.appendChild(node); // node is appended before click event is dispatched
        node.dispatchEvent(new MouseEvent('click'));
    });   
}
```

This change fixes the issue on Safari on iOS 10 (fixing both 'Take Photo' and 'Photo Library'). This continues to work on Chrome on Android, and Chrome on Linux without any issue.

**Comment by evancz** [comment]
This _should_ be fixed in `1.0.2` which uses the same backing `<input>` node for each command. Assuming the root problem is that some browsers GC even though the function may still be called, this resolves that by keeping the `<input>` alive in the global scope of the program.

Based on the GC theory, I made a program that will allocate a bunch of models very quickly to try to get GC to run more often. The result was:

- [Example 1.0.1](https://ellie-app.com/4BrBCtFgYyPa1) which fails pretty reliably in Safari 12, especially if you wait a bit on the file selection dialog.
- [Example 1.0.2](https://ellie-app.com/4BrJpDvGyx2a1) which I could not get to fail in Safari 12 or FireFox 64.

Thanks for the report, and thank you @charukiewicz for sharing some ideas on this on discourse! If you try this out in other browsers, please share the results on the discourse thread: https://discourse.elm-lang.org/t/cross-browser-compatibility-fix-for-file-select-file-s-in-elm-file/3060/5
